src/tools/openapi.ts

```diff
@@ -124,15 +155,18 @@ Only use the OpenAPI tool if you need to communicate to external servers.`;
     if (!this.openApiSchema?.paths) {
       throw new ValueError("Server is not specified!");
     }
+    this.url = options.url;
+    this.apiKey = options.apiKey;
+    this.httpProxyUrl = options.httpProxyUrl;
```

Comment:
Wrong. You don't need to set those properties because they are already set in `this.options`